src/features/book/bookSlice.js

```diff
@@ -0,0 +1,54 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/WA5eot1mOTs5Y1g1bAoc/books';
+
+const initialState = {
+  booksByItem: {},
+  isLoading: true,
+  error: null,
+};
+
+export const getBookItems = createAsyncThunk(
+  'book/getBookItems',
+  async (thunkAPI) => {
+    try {
+      const resp = await axios(url);
+      return resp.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('something went wrong');
+    }
+  },
+);
```

Comment:
#### Hello @zhorabay, 👋

- [ ] First of all, kudos on your progress in reaching the API in this milestone! 🎉 It's evident that you've put in some solid work. 🚀 However, there's a minor hiccup that's causing your UI to render only a 'loading state'. 🤔

  - The issue lies in the fact that you've beautifully set up your `getBookItems` function using `createAsyncThunk`, but you might have missed dispatching it in your `Bookcomponents.js`. As a result, the data from your API isn't making its way to the component for display, hence the perpetual 'loading state'.

   - No worries, though! 🛠️ This is a common pitfall, and it happens to the best of us. To resolve this, simply make use of `useDispatch` to dispatch `getBookItems` within `Bookcomponents.js`. This should get those books flowing into your component, and you'll have a working UI in no time! 📚👀

Happy coding! 😊🚀🍀